src/library.h

```diff
@@ -1,6 +1,10 @@
 #ifndef LIBRARY_H_INCLUDED
 #define LIBRARY_H_INCLUDED
 
+#ifdef __cplusplus
+extern "C" {
+#endif

```

Comment:
Same comment as above applies here as well. The #include and struct Library do not need be be part of extern "C" {} clause.